sound/soc/sof/sof-client.c

```diff
@@ -284,6 +285,20 @@ int sof_client_ipc_set_get_data(struct sof_client_dev *cdev, void *ipc_msg,
 }
 EXPORT_SYMBOL_NS_GPL(sof_client_ipc_set_get_data, SND_SOC_SOF_CLIENT);
 
+struct sof_ipc4_fw_module *sof_client_ipc4_find_module_by_uuid(struct sof_client_dev *cdev, const guid_t *uuid)
+{
+	struct snd_sof_dev *sdev = cdev->sdev;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_IPC4)
+	if (sdev->pdata->ipc_type == SOF_INTEL_IPC4)
+		return sof_ipc4_find_module_by_uuid(sdev, uuid);
+#endif
+	dev_err(sdev->dev, "Only supported with IPC4\n");
+
+	return NULL;
```

Comment:
We have a Kconfig option for compiling in IPC3 support, IPC4 support, or both, then we have a private data variable telling which one should be used.  So its possible IPC4 not built in, in which case the #if is necessary to avoid undefined symbol. But its also possible that IPC4 is built, but it just is not in use (depends on the module parameters). Then again the sof_client API is common for both IPC modes, but this particular function does not make any sense in IPC3 mode. I do not know any better way to deal with the situation.